substrace_lints/src/substrace_lints/enable_singlepass_benchmarks.rs

```diff
@@ -0,0 +1,107 @@
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint, impl_lint_pass};
+use serde::{Deserialize, Serialize};
+use std::str;
+
+use grep::printer::JSON;
+use grep_pcre2::RegexMatcher;
+use grep::searcher::{BinaryDetection, SearcherBuilder};
+use std::error::Error;
+use std::io::BufWriter;
+use walkdir::WalkDir;
+
+declare_lint! {
+    pub ENABLE_SINGLEPASS_BENCHMARKS,
+    Warn,
+    "TODO"
+}
+
+impl_lint_pass!(EnableSinglepassBenchmarks => [ENABLE_SINGLEPASS_BENCHMARKS]);
+
+#[derive(Clone, Default)]
+pub struct EnableSinglepassBenchmarks;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct MyGrepResult {
+    #[serde(rename = "type")] 
+    type_name: String,
+    data: serde_json::Value,
+}
+
+//TODO: Implement auto-fix.
+impl<'tcx> LateLintPass<'tcx> for EnableSinglepassBenchmarks {
+
+    fn check_crate(&mut self, cx: &LateContext<'tcx>) {
+
+        //TODO: Disregard patterns matched in block comments
+        // See this pattern in action: https://regex101.com/r/jd6CX1/4
```

Comment:
I don't mind this catching errors in comments; that enforces best practices in comments too.